### DISCUSSION TITLE
希望config生效在on_load之前

### DISCUSSION THREAD

**Discussion by PeakRacing** [xmake-io/xmake#2920] [Ideas]
### 你在什么场景下需要该功能？

你好我希望通过输入命令f --tar=abc动态添加静态库依赖，在on_load 里面走 target:add("xxx", ) 动态添加，但是此时config还未生效，has_config("tar")获取的是nil，可否将config生效提前到on_load之前？

### 描述可能的解决方案

将config生效提前到on_load之前

**Comment by waruqi** [comment]
不能，on_load 是最早的，没法改，最早只能在 on_config 里判断 has_config。。不过添加 target deps 只能在 on_load/after_load 做，on_config 里不确定，可能可以 也可能不行

**Reply by PeakRacing** [reply]
我试了下on_config里走target:add不行，有没有其他的办法可以实现呢？或者在on_config 里可以实现此操作

**Reply by waruqi** [reply]
不能 deps 关系必须在早期确定好，后面很多地方都要用到，你通过 config 去动态改，本身就已经很滞后了。。
